src/build.d

```diff
@@ -1441,6 +1450,28 @@ auto log(T...)(T args)
         writefln(args);
 }
 
+/**
+Aborts the current build and constructs a BuildException to  be thrown by the caller
+
+TODO:
+    - Display detailed error messages
+    - Handle spawned processes
+
+Params:
+    msg = error message to display
+
+Returns: a BuildException containing msg
+*/
+BuildException abortBuild(string msg = "Build failed!")
+{
+    return new BuildException(msg);
```

Comment:
You could also throw here.  it's looks weird to throw both at the callsite and inside the function, but doing both has benefits even if the caller's `throw` never actually gets executed.

Reply:
Indeed, double throw would be really weird. How would that be beneficial?

Reply:
someone might accidently call `abortBuild` without `throw abortBuild`, thinking that it would exit the program (since it's called `abortBuild`).  Since error conditions like this are rarely tested, it's better to make these code paths as "full proof" as you can.

If you throw in both places, the caller can choose to use the explicit `throw` or not, depending on if they need that control flow.  Take this example
```
int returnInt()
{
    if (foo)
    {
        return 0;
    }
    abortBuild();
}
```

In this case you would get a compile error because the end of the function doesn't return anything.  However, if `abortBuild` returns an Exception type even though it never returns, then you can just do `throw abortBuild()` and the compiler will be happy.

Note that there's been some discussion about adding a trait or a special type that means "noreturn", but D doesn't have that yet.   This is sort of a workaround for not having that feature.

Reply:
This can be an easy mistake to make as well:
```D
void foo()
{
    if (somethingWentWrong)
        abortBuild("oh no");
    okWereGood();
}
```

If you throw in both places, then this works as one would expect from just looking at the code without going into the function definitions.

If you don't want to throw inside `abortBuild`, then it's better to call it something like `makeBuildError`, as it's not really aborting the build, just returning an error that you need to throw yourself.

Reply:
Clever. I'll revert to throwing inside of `abortBuild` but keep the fake return value to allow `throw abortBuild`at the callsite